src/equations/linear_scalar_advection_1d.jl

```diff
@@ -202,14 +202,16 @@ end
 
 @inline function splitting_lax_friedrichs(u, ::Val{:plus}, orientation::Integer,
                                           equations::LinearScalarAdvectionEquation1D)
+    RealT = eltype(u)
     a = equations.advection_velocity[1]
-    return a > 0 ? flux(u, orientation, equations) : zero(u)
+    return a > 0 ? flux(u, orientation, equations) : zero(RealT)
 end
 
 @inline function splitting_lax_friedrichs(u, ::Val{:minus}, orientation::Integer,
                                           equations::LinearScalarAdvectionEquation1D)
+    RealT = eltype(u)
     a = equations.advection_velocity[1]
-    return a < 0 ? flux(u, orientation, equations) : zero(u)
+    return a < 0 ? flux(u, orientation, equations) : zero(RealT)
```

Comment:
I think this change is problematic since `u::SVector{1, Float64}` has a different type than `zero(Float64)`.

Reply:
Good point! Thanks!